sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/Changefeed.java

```diff
@@ -100,25 +116,24 @@
     private Mono<OffsetDateTime> populateLastConsumable() {
         /* We can keep the entire metadata file in memory since it is expected to only be a few hundred bytes. */
         return DownloadUtils.downloadToByteArray(this.client, METADATA_SEGMENT_PATH)
+            .flatMap(DownloadUtils::parseJson)
             /* Parse JSON for last consumable. */
-            .flatMap(json -> {
-                try {
-                    JsonNode jsonNode = MAPPER.reader().readTree(json);
-                    this.lastConsumable = OffsetDateTime.parse(jsonNode.get("lastConsumable").asText());
-                    if (this.lastConsumable.isBefore(endTime)) {
-                        this.safeEndTime = this.lastConsumable;
-                    }
-                    return Mono.just(this.lastConsumable);
-                } catch (IOException e) {
-                    return FluxUtil.monoError(logger, new UncheckedIOException(e));
+            .flatMap(jsonNode -> {
+                /* Last consumable time. The latest time the changefeed can safely be read from.*/
+                OffsetDateTime lastConsumableTime = OffsetDateTime.parse(jsonNode.get("lastConsumable").asText());
+                /* Soonest time between lastConsumable and endTime. */
+                OffsetDateTime safeEndTime = this.endTime;
+                if (lastConsumableTime.isBefore(endTime)) {
```

Comment:
endTime is/should be optional parameter. How do we handle that case? I.e. we should default to lastConsumable if end time is null . Also if end time is null we shouldn't populate it in the cursor.

Reply:
I think ChangefeedFactory deals with this - we default to the Max time

OffsetDateTime start = startTime == null ? OffsetDateTime.MIN : startTime;
        OffsetDateTime end = endTime == null ? OffsetDateTime.MAX : endTime;

        return new Changefeed(this.client, start, end, null, segmentFactory);

Reply:
I feel like if endTime is null, the user just wants to go forever so I don't see the issue in doing this

Reply:
great. sounds good.